scylla-cql/src/types/deserialize/value.rs

```diff
@@ -66,6 +66,75 @@ where
     }
 }
 
+macro_rules! impl_strict_type {
+    ($cql_name:literal, $t:ty, $cql_type:pat, $conv:expr $(, $l:lifetime)?) => {
+        impl<$($l,)? 'frame> DeserializeCql<'frame> for $t
+        where
+            $('frame: $l)?
+        {
+            fn type_check(typ: &ColumnType) -> Result<(), ParseError> {
+                // TODO: Format the CQL type names in the same notation
+                // that ScyllaDB/Casssandra uses internally and include them
+                // in such form in the error message
```

Comment:
What notation do you have in mind? Is there some list of those types in the notation you and the comment mention?

Reply:
I'm not sure what notation the comment has in mind, as it is not my PR - I just think that if this is something that should be done here then it should probably be done in serialization too because it also formats CQL types.

Reply:
@piodul Could you shed some light here?

Reply:
I think the "notation used internally" was just supposed to mean "CQL syntax". I think it's a good UX if the error messages display the types in CQL syntax.

Reply:
Is the  syntax currently displayed in the code in line with the CQL syntax (as you mean it)?

Reply:
I have no idea :) I didn't touch the rust driver code for a while. You can write an example that triggers this error and see for yourself.

Reply:
So from what I can see, with simple types (non-collections) our naming differs only in being non-lowercase. In case of collections, though, it may be a bit more complicated. I'll investigate it further, but is it really a big deal to have the types displayed in the same case as the specification?

Reply:
I don't think this is a big deal as long as it is possible to understand which CQL type the error message is about. It's definitely not something that would block the merge, I think that's why I left a todo there.

However, from the UX perspective, I think it's worth it - I suppose that the CQL syntax will be most familiar to users, so it would be nice to implement this at some point.

Reply:
OK, I consider this out of scope of this PR. In the future we could make a code-wide change to use CQL syntax everywhere.